src/main/com/fulcrologic/rad/form.cljc

```diff
@@ -592,15 +592,19 @@
 
       This mutation's ok-action will also update the data in the local state."
      [{:keys [root-ident entity delta]}]
-     (ok-action [{:keys [state]}]
+     (ok-action [{:keys [state tempid->realid]}]
        (if delta
-         (doseq [[ident changes] delta
+         (doseq [[ident changes] (tempid/resolve-tempids delta tempid->realid)
                  :let [data-to-merge (reduce-kv
                                        (fn [m k v] (assoc m k (:after v)))
                                        {}
                                        changes)]]
            (swap! state update-in ident merge data-to-merge))
-         (swap! state update-in root-ident merge entity)))
+         (swap! state
```

Comment:
I don't have time to test this. I assume you did.  There are a few things about this that probably are not quite right...the after could be nil, which means dissoc instead of assoc (my bad)...but it's good enough I guess if you're just trying to get tempids right.